### DISCUSSION TITLE
Dynamic prompt not changing between images within batch

### DISCUSSION THREAD

**Discussion by MrUrzza** [adieyal/sd-dynamic-prompts#475] [Q&A]
When running a batch generation, only a single prompt is generated using my variables, but then its used for all my images. For example, when running the prompt `safe, a {dog|cat|horse} with {red|blue|green} fur, simple background` with a batch count of 9, I get nine blue horses, instead of a mix of animals and colors. Rerunning the task will give me nine new animals of a different color, but they still will all be the same thing. (ex, each will be some variation of a green cat.)

Ive tried checking an unchecking most boxes, so I dont think its a settings thing. Any ideas? I'm on ver. 1.3.0 of Automatic1111

Dynamic prompt is working for Vlad 1111

Troubleshooting steps so far
*Deleted the dynamic prompts folder from automatic and reinstalling it
*Checking and unchecking various settings boxes
*Updating automatic
*Checking for updates for dynamic prompts
*Testing if the issue also occurs when using batch size and depth; it does

**Comment by MrUrzza** [comment]
![grid-0029](https://github.com/adieyal/sd-dynamic-prompts/assets/134964094/2f535162-b18b-4a15-90de-3e61ade089e7)
An example of the output im getting

**Comment by M-Walton** [comment]
I have the same issue - img2img batch processing, everything uses the same random prompt on any given run.
I would like an option to generate a fresh dynamic prompt for each image processed in the batch. (Ideally with the option to output txt files noting the generated prompt, but that's a side thing)

**Comment by MrUrzza** [comment]
Issue no longer occurs. Suspect it may have been a negative interaction with unprompted.
